@here/harp-mapview/lib/MapView.ts

```diff
@@ -3417,13 +3379,13 @@ export class MapView extends EventDispatcher {
      * Update `Env` instance used for style `Expr` evaluations.
      */
     private updateEnv() {
-        this.m_env.entries.$zoom = this.m_zoomLevel;
+        this.m_tileObjectRenderer.env.entries.$zoom = this.m_zoomLevel;
 
         // This one introduces unnecessary calculation of pixelToWorld, even if it's barely
         // used in our styles.
-        this.m_env.entries.$pixelToMeters = this.pixelToWorld;
+        this.m_tileObjectRenderer.env.entries.$pixelToMeters = this.pixelToWorld;
 
-        this.m_env.entries.$frameNumber = this.m_frameNumber;
+        this.m_tileObjectRenderer.env.entries.$frameNumber = this.m_frameNumber;
```

Comment:
There seem to be a few cases where MapView interacts with MapEnv so I wonder if it doesn't make more sense to keep it in MapView and just pass it to the `TileObjectRenderer` ?

Otherwise move `updateEnv` to the `TileObjectRenderer`, so the classes try to be as encapsulated as possible.

Reply:
yeah, good point, I also saw that it broke some tests, I ll review that

Reply:
I moved the MapEnv back to MapView